### ISSUE TITLE
Main Menu/Options Menu Adjustments

### ISSUE THREAD

**Issue by Polymega** [elishacloud/Silent-Hill-2-Enhancements#275]
Hi @belek666,

-------------------

### Options menu adjustment

**Update:** Aero_ found a solution to this request: https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/199#issuecomment-623694764

~~In the options menu for the game, for "Weapon Control" and "View Control", the two selections are 'Normal' and 'Switch'.~~

~~Inside `option_msg_*.mes` there is an unused string that says "Press".~~

![image](https://user-images.githubusercontent.com/27395260/81010351-25a8ae80-8e24-11ea-842a-3c9efcbc3575.png)

~~Can the selections for "Weapon Control" and "View Control" be changed to 'Press' and 'Switch'?
"Press" should have been the correct selection this whole time.~~

Thank you to Badore90 for finding the unused "Press" string for use here.

--------------------------------

### Main menu adjustment

This is in regards to the `MainMenuTitlePerLang` feature.

If you enable the Japanese language selection (`UnlockJapLang = 1`) and select the Japanese language in the options menu, the main menu will then look like this:

![image](https://user-images.githubusercontent.com/27395260/81009728-25f47a00-8e23-11ea-93cb-b1e3a98e3d57.png)

... unless you have a `start01j.tex` file present in `SILENT HILL 2\sh2\pic\etc\`. 

You created a failsafe check that, if the correct `start01*.tex` is **not** present, it will default to using `start01.tex` instead. I believe this safety check might have been excluded for the Japanese language selection. [Possibly in this portion of code?](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/master/Patches/MainMenu.cpp#L65-L74)

**Comment by belek666** [comment]
Hello @Polymega 
The problem does not exist when not using hi res menu textures. By looking at texture patching code in `TexPatch.cpp` I can say that `start01.tex` resolution is read only once at start and texture scale values are calculated at this time. Solution would be to update those variables every Main Menu call. The `ScaleStart01Texture` function (with small modification - writing jumps to ASM code only once) could be called in function `UpdateTitlePath` in `MainMenu.cpp`.

**Comment by Polymega** [comment]
Thank you, Belek. Unfortunately, I still have the error.

Can I have you double-check to ensure this isn't an issue with `MainMenuTitlePerLang`? When using the low res textures (`EnableTexAddrHack = 0`), there is still display issues when switching to the Japanese language (`UnlockJapLang = 1`) whenever no `start01j.tex` exists:

![image](https://user-images.githubusercontent.com/27395260/81120926-a8e90380-8efb-11ea-9ba8-20157a285b21.png)

I just want to be sure we're addressing this error at the right area.

Again, there is a safety check to default to `start01.tex` if a `start01*.tex` file does not exist, but it appears this safety check may not correctly happen if a `start01j.tex` file does not exist?

**Comment by Polymega** [comment]
Wait a minute. I think this is correct. I forgot the game has a default `start01j.tex` file (that comes with the vanilla game).

Comparing the original two textures:
![Untitled-2](https://user-images.githubusercontent.com/27395260/81121451-cec2d800-8efc-11ea-8d71-c07add6ad960.gif)

So the game inherently has code to switch to `start01j.tex` whenever the Japanese language is selected. But the PC version of the game was never meant to select the Japanese language.

To fix this, we just need to ensure a high resolution `start01j.tex` file accompanies all the other high res files, because their exists code already in the game to switch to this texture if the Japanese language were to ever be selected.

Sorry for the confusion on this, but I'm glad we discussed this out loud. I think everything is fine/safe as it currently is, in regards to the main menu!

**Comment by Polymega** [comment]
I deleted my two posts from a moment ago. I got myself all sorts of confused. 

Even though the vanilla game comes with `start00j.tex` and `start01j.tex` files, it will not inherently switch over to the Japanese equivalents if you select the Japanese language (`UnlockJapLang = 1`). 

Your `MainMenuTitlePerLang` was doing that, like it does for all the other languages. But because we didn't have a high resolution version of `start01j.tex`, it was causing the visual error due to the resolution differences of switching back to the low res `start01j.tex`.

Your proposed fix will check for this resolution difference on each language selection switch. I understand now. Sorry for the confusion. A lot of different variables at play within the INI file to test this! Your fix worked well!
